code/model/jobs/DNDeployment.php

```diff
@@ -197,6 +197,66 @@ public function getTags() {
 	}
 
 	/**
+	 * Collate the list of additional flags to affix to this deployment.
+	 * Elements of the array will be rendered literally.
+	 *
+	 * @return ArrayList
+	 */
+	public function getFullDeployMessages() {
+		$strategy = $this->getDeploymentStrategy();
+		if ($strategy->getActionCode()!=='full') return null;
+
+		$changes = $strategy->getChangesModificationNeeded();
+		$messages = [];
+		foreach ($changes as $change => $details) {
+			if ($change==='Code version') continue;
+
+			$messages[] = [
+				'Flag' => sprintf(
+					'<span class="label label-default full-deploy-info-item">%s</span>',
+					$change[0]
+				),
+				'Text' => sprintf('%s changed', $change)
+			];
+		}
+
+		if (empty($messages)) {
+			$messages[] = [
+				'Flag' => '',
+				'Text' => '<i>Environment changes have been made.</i>'
+			];
+		}
+
+		return new ArrayList($messages);
+	}
+
+	/**
+	 * Get bootstrape-y class for the frontend.
+	 *
+	 * @return string
+	 */
+	public function getStatusClass() {
+		switch ($this->Status) {
+			case 'Finished': return 'success';
+			case 'Failed': return 'danger';
+			case 'n/a': return 'warning';
+			default: return 'info';
+		}
+	}
+
+	/**
+	 * Convert status to a friendly form, shown on the frontend.
+	 *
+	 * @return string
+	 */
+	public function getStatusFriendly() {
```

Comment:
I removed the code because it doesn't belong in this commit :P ready for re-review.